### DISCUSSION TITLE
Season 1 of new ideas.

### DISCUSSION THREAD

**Discussion by reddec** [reddec/trusted-cgi#14] [Ideas]
Please share a new ideas which you would like to see implemented in trusted-cgi.

Rules:

- Doable 😄 
- Will keep positioning of the project: do not consume resource unless there is a job (CGI).

I would like to do it as "seasons", maybe twice a year.

**Comment by c-nv-s** [comment]
Support for reading files from s3 could be useful

**Comment by c-nv-s** [comment]
Export lambda usage report functionaliy (daily/weekly/monthly)

**Comment by c-nv-s** [comment]
on failure of some lambdas, allow a header to be sent back to client indicating time after which to retry the call (like the Retry-After header in Hasura https://hasura.io/docs/latest/event-triggers/payload/#retry-after-header )

**Comment by c-nv-s** [comment]
support triggering of lambdas by listening/subscribing to messages from a message broker like https://github.com/nats-io/nats-server
